packages/standard-minifier-js/plugin/minify-js.js

```diff
@@ -1,105 +1,254 @@
-          let currentLine = lines[i];
+// Registrar o minificador apenas quando o Plugin estiver disponível (não nos testes)
+if (typeof Plugin !== 'undefined') {
+  Plugin.registerMinifier({
+      extensions: ['js'],
+      archMatching: 'web',
+    },
+    () => new MeteorMinifier()
+  );
+}
 
-          // If the line is a boatload of slashes (8 or more), we're in the right place.
-          if (/^\/\/\/{6,}$/.test(currentLine)) {
+export class MeteorMinifier {
 
-              // If 4 lines back is the same exact line, we've found the framing.
-              if (lines[i - 4] === currentLine) {
+  constructor() {
+    this.config = getConfig();
+  }
 
-                  // So in that case, 2 lines back is the file path.
-                  let originalFilePath = lines[i - 2].substring(3).replace(/\s+\/\//, "");
+  _minifyWithSWC(file) {
+    swc = swc || require('@meteorjs/swc-core'); 
+    const NODE_ENV = process.env.NODE_ENV || 'development';
+    
+    let content = file.getContentsAsString();
+
+
+    return swc.minifySync(
+      content,
+      {
+        ecma: 5,
+        compress: {
+          drop_debugger: false,
+
+          unused: true,
+          dead_code: true,
+          typeofs: false,
+
+          global_defs: {
+            'process.env.NODE_ENV': NODE_ENV,
+          },
+        },
+        safari10: true,
+        inlineSourcesContent: true
+      }
+    );
+  }
 
-                  throw new Error(
-                      `terser minification error (${error.name}:${error.message})\n` +
-                      `Source file: ${originalFilePath}  (${originalSourceFileLineNumber}:${error.col})\n` +
-                      `Line content: ${lineContent}\n`);
-              }
-          }
-          originalSourceFileLineNumber++;
+  _minifyWithTerser(file) {
+    let terser = require('terser');
+    const NODE_ENV = process.env.NODE_ENV || 'development';
+    const content = file.getContentsAsString();
+    
+    return terser.minify(content, {
+      compress: {
+        drop_debugger: false,
+        unused: false,
+        dead_code: true,
+        global_defs: {
+          "process.env.NODE_ENV": NODE_ENV
+        }
+      },
+      // Fix issue meteor/meteor#9866, as explained in this comment:
+      // https://github.com/mishoo/UglifyJS2/issues/1753#issuecomment-324814782
+      // And fix terser issue #117: https://github.com/terser-js/terser/issues/117
+      safari10: true
+    }).then(result => {
+      if (!result) {
+        throw new Error(`Terser produced empty result for ${file.getPathInBundle()}`);
       }
+      return result;
+    }).catch(error => {
+      throw error;
+    });
+  }
+
+  minifyOneFile(file) {
+    // TODO: create a function to check if the file should be skipped and share it with babel-compiler.js file
+    
+    // Legacy config check - kept for backwards compatibility
+    if(this.config && this.config.modernTranspiler === false) {
+      return this._minifyWithTerser(file);
```

Comment:
We don't need to have this backwards compatibility check since this was introduced in a beta version, we just switched to `modern`.

Reply:
But for minifier?

Reply:
modernTranspiler no longer exist for upcoming beta-1. So this is not needed.